app/services/unsuspend_account_service.rb

```diff
@@ -16,6 +20,22 @@ def unsuspend!
     @account.unsuspend! if @account.suspended?
   end
 
+  def refresh_remote_account!
+    return if @account.local?
+
+    # While we had the remote account suspended, it could be that
+    # it got suspended on its origin, too. So, we need to refresh
+    # it straight away so it gets marked as remotely suspended in
+    # that case.
+
+    @account.update!(last_webfingered_at: nil)
+    @account = ResolveAccountService.new.call(@account)
+
+    # Worth noting that it is possible that the remote has not only
+    # been suspended, but deleted permanently, in which case
+    # @account would now be nil.
+  end
+
   def merge_into_home_timelines!
     @account.followers_for_local_distribution.find_each do |follower|
       FeedManager.instance.merge_into_timeline(@account, follower)
```

Comment:
Note that `merge_into_timeline` does not exist. This is a bug from the latest FeedManager refactoring, In a similar way, `unmerge_from_timeline` does not exist either. (#15099)